app/src/main/java/com/github/sdp_begreen/begreen/fragments/ProfileDetailsFragment.kt

```diff
@@ -0,0 +1,105 @@
+package com.github.sdp_begreen.begreen.fragments
+
+import android.os.Build
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.ProgressBar
+import android.widget.RatingBar
+import android.widget.TextView
+import androidx.lifecycle.lifecycleScope
+import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.models.Actions
+import com.github.sdp_begreen.begreen.models.User
+import kotlinx.coroutines.launch
+
+
+
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [ProfileDetailsFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class ProfileDetailsFragment : Fragment() {
+    var user: User? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            user = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                it.getParcelable(ARG_USER, User::class.java)
+            }else {
+                it.getParcelable(ARG_USER)
+            }
+        }
+
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+
+        val view = inflater.inflate(R.layout.fragment_profile_details, container, false)
+        val name = view.findViewById(R.id.fragment_profile_details_profile_name) as TextView
+        val rating = view.findViewById(R.id.fragment_profile_details_profile_rating) as RatingBar
+        val profileImgView = view.findViewById(R.id.fragment_profile_details_profile_image) as ImageView
+        val profileDescription = view.findViewById(R.id.fragment_profile_details_profile_description) as TextView
+        val profilePhone = view.findViewById(R.id.fragment_profile_details_profile_phone) as TextView
+        val profileEmail = view.findViewById(R.id.fragment_profile_details_profile_email) as TextView
+        val userTextLevel= view.findViewById(R.id.fragment_profile_details_level) as TextView
+        val userProgressBar = view.findViewById(R.id.fragment_profile_details_user_progress) as ProgressBar
+        val followButton = view.findViewById(R.id.fragment_profile_details_follow_button) as Button
+
+        name.text = user?.name
+        rating.rating = user?.score?.toFloat() ?: 0.0f
+        profileImgView.setImageBitmap(user?.img?.getPhotoFromDataBase())
+        profileDescription.text = user?.description
+        profilePhone.text = user?.phone
+        profileEmail.text = user?.email
+        userTextLevel.text = user?.name + "'s level"
```

Comment:
Here, We should not set the value by hand, we should get it from the `string.xml` file to provide better integration and support for multiple language easily later.

Reply:
Yep not a bad habit to take for multilingual purpose. Thx!